specification.md

```diff
@@ -5187,71 +5689,35 @@ Note the same can be also specified using workflow metadata, which is the prefer
 
-Runtime implementations must be able to map the error domain specific name (and the optional error code)
-to concrete technical errors that arise during workflow execution.
+#### Error Source
 
-#### Defining Errors
+In Serverless Workflow, the concept of "Error Source" refers to the precise origin or location within the workflow definition where an error occurs during its execution. This crucial aspect is identified and pinpointed using the [error definition's ](#error-definition) `instance` property, which is defined as an [RFC6901 JSON pointer](https://datatracker.ietf.org/doc/html/rfc6901).
 
-Known workflow errors, that we know we need to handle during workflow execution should be defined in
-the workflow top-level 'errors' property. This property can be either a string type, meaning it can reference 
-a reusable JSON or Yaml definition file including the error definitions, or it can have an array type where you can
-define these checked errors in-line in your workflow definition.
+When an error arises during the execution of a workflow, whether it's at the level of an action or a state, the Error Source becomes instrumental in identifying the specific component within the workflow where the error originated. This granular identification is essential for efficient debugging and troubleshooting, as it allows developers to swiftly locate and address the root cause of the error.
 
-Here is an example of such a definition for both cases:
+By leveraging the Error Source, developers can streamline the error-handling process, facilitating quicker resolution of issues and enhancing the overall reliability and robustness of the workflow.
 
-1. Referencing a reusable JSON/Yaml error definition file:
+#### Error Handling Strategies
 
-<table>
-<tr>
-    <th>JSON</th>
-    <th>YAML</th>
-</tr>
-<tr>
-<td valign="top">
+In ServerlessWorkflow, you have the flexibility to define error handling strategies using error handlers, policies, and outcome definitions.
 
-```json
-{
-"errors": "file://documents/reusable/errors.json"
-}
-```
+Errors can be configured at both the state and action levels, allowing you to tailor error handling to specific components within your workflow.
 
-</td>
-<td valign="top">
+When choosing an error handling strategy, consider your workflow requirements and strike a balance between simplicity, maintainability, and flexibility. Choose the approach that best fits the needs of your workflow to ensure effective error management.
 
-```yaml
-errors: file://documents/reusable/errors.json
-```
+##### Inline Error Handling
 
-</td>
-</tr>
-</table>
-
-2. Defining workflow errors in-line:
+The most basic method involves configuring the onErrors property directly within a state or an action and adding an inline handler. While suitable for specific scenarios, this approach should be used sparingly as it may lead to code duplication and reduced maintainability.
```

Comment:
```suggestion
The most basic method involves configuring the `onErrors` property directly within a state or an action and adding an inline handler. While suitable for specific scenarios, this approach should be used sparingly as it may lead to code duplication and reduced maintainability.
```